model/src/main/kotlin/utils/CompositePackageConfigurationProvider.kt

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (C) 2022 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model.utils
+
+import org.ossreviewtoolkit.model.Identifier
+import org.ossreviewtoolkit.model.Provenance
+import org.ossreviewtoolkit.model.config.PackageConfiguration
+
+/**
+ * A [PackageConfigurationProvider] that combines the provided [providers] into a single provider. The order of the
+ * [providers] determines the order in which they are queried when requesting a [PackageConfiguration] in
+ * [getPackageConfiguration].
+ */
+class CompositePackageConfigurationProvider(
+    private val providers: List<PackageConfigurationProvider>
+) : PackageConfigurationProvider {
+    constructor(vararg providers: PackageConfigurationProvider) : this(providers.toList())
```

Comment:
Sorry, one more nit: `toList` could become `asList` for efficiency. But this could also be done as a follow-up.